### PR TITLE
remove Util methods

### PR DESCRIPTION
**Pull request by gr2m**
only Util.log left, I willll probably replace that with npm.im/debug in future

### CODE REVIEW

File: lib/error.js [hunk 1]
```diff
@@ -105,13 +105,7 @@ for (var status in statusCodes) {
 
   Util.inherits(error, exports.HttpError)
 
-  var className = toCamelCase(defaultMsg)
+  var className = defaultMsg.replace(/\s/g, '')
```
Comment (bkeepers) [inline lib/error.js:108]: This looks like it will change functionality. What are the ramifications?

**Review by bkeepers — COMMENTED** [review]
> only Util.log left, I willll probably replace that with npm.im/debug in future

I'd greatly appreciate if the logger could be made configurable. For example, Probot uses [bunyan](https://www.npmjs.com/browse/keyword/bunyan), which has cool features for logging and formatting objects so you can query logs by different properties.

File: lib/error.js (hunk 1 above)
Comment (gr2m) [reply lib/error.js:108]: In the context of this file, the `toCamelCase` method just turned messages like `Requested Range Not Satisfiable` into an Error name like `RequestedRangeNotSatisfiable`. It used its own toCamelCase method, not the Util.toCamelCase one

https://github.com/octokit/node-github/blob/fb0ad883f6f79007d41f6a94258c9b8a09fc2454/lib/error.js#L113-L117

I've double checked that, it's all good

**Comment by gr2m** [comment]
@bkeepers 

> I'd greatly appreciate if the logger could be made configurable 

we have to take into account that bundle size for browser usage is a concern. It should be fairly simple to add custom logging with pre/post request hooks as needed, I'd keep the core one minimal

**Comment by bkeepers** [comment]
> we have to take into account that bundle size for browser usage is a concern

Just allowing me to inject a logger object in the constructor that conforms to the `console` interface would be ideal.

**Pull request merged**